core/src/test/java/org/kohsuke/stapler/export/JSONDataWriterTest.java

```diff
@@ -18,6 +18,31 @@
         return w.toString();
     }
 
+    @Test
+    public void testJsonCharacters() throws Exception {
+        String jsonCharactersExpected = "{\"_class\":\"JsonCharacters\",\"value\":\"\\\"'\\/[]{}\\\\\"}";
+        assertEquals(jsonCharactersExpected, serialize(new JsonCharacters(), JsonCharacters.class));
+
+        final String jsonCharacters2Expected = "{\"_class\":\"JsonCharacters2\",\"value\":\"\\\"foo\\\" isn't \\\"bar\\\". specials: \\b\\r\\n\\f\\t\\\\\\/\"}";
+        assertEquals(jsonCharacters2Expected, serialize(new JsonCharacters2(), JsonCharacters2.class));
+    }
+
+    @ExportedBean
+    public static class JsonCharacters {
+        @Exported
+        public String getValue() {
+            return "\"'/[]{}\\";
+        }
+    }
+
+    @ExportedBean
+    public static class JsonCharacters2 {
+        @Exported
+        public String getValue() {
+            return "\"foo\" isn't \"bar\". specials: \b\r\n\f\t\\/";
```

Comment:
I'm not seeing any tests covering the change this brings to escaping Unicode characters (specifically the method you are using will use the `\u0080` for any codepoint above 0x80... not strictly necessary for JSON as that should be in a UTF encoding always